modules/raft/src/main/java/org/apache/ignite/raft/jraft/util/concurrent/ConcurrentLinkedFifoByteBufferCollectorPool.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.raft.jraft.util.concurrent;
+
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.ignite.raft.jraft.util.ByteBufferCollector;
+import org.apache.ignite.raft.jraft.util.ByteBufferCollectorPool;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Thread safe implementation of a pool based on a fixed-size FIFO linked list.
+ *
+ * <p>{@link ByteBufferCollector} are given out on a first-in, first-out basis.</p>
+ */
+public class ConcurrentLinkedFifoByteBufferCollectorPool implements ByteBufferCollectorPool {
+    private final int capacity;
+
+    private final AtomicReference<Node> stack = new AtomicReference<>();
+
+    /**
+     * Constructor.
+     *
+     * @param capacity Maximum capacity of the pool, expected to be greater than zero.
+     */
+    public ConcurrentLinkedFifoByteBufferCollectorPool(int capacity) {
+        assert capacity > 0 : capacity;
+
+        this.capacity = capacity;
+    }
+
+    /** Removes and returns the last added {@link ByteBufferCollector}, {@code null} if the pool is empty. */
+    @Override
```

Comment:
Do we really need javadocs on overrides?

Reply:
I think it's worth it, he clarifies the method contract to make it clearer.